parameter.go

```diff
@@ -0,0 +1,178 @@
+package taskflow
+
+import (
+	"errors"
+	"strconv"
+)
+
+type parameter struct {
+	info     ParameterInfo
+	newValue func() Value
+}
+
+// ParameterInfo represents the general information of a parameter for one or more tasks.
+type ParameterInfo struct {
```

Comment:
maybe let's rename it to `ParamInfo` for consistency?

Reply:
I will skip it for now. I have an idea of how to redesign the parameter registration. I want to make a draft PR for it first.

Reply:
Refactored 😄